lib/checkautovariables.cpp

```diff
@@ -58,7 +58,7 @@ static bool isPtrArg(const Token *tok)
 static bool isArrayArg(const Token *tok, const Settings* settings)
 {
     const Variable *var = tok->variable();
-    return (var && var->isArgument() && var->isArray() && !settings->library.isentrypoint(var->scope()->className));
+    return (var && var->isArgument() && var->isArray() && var->scope() && !settings->library.isentrypoint(var->scope()->className));
```

Comment:
Should we even create a variable without a scope?

Reply:
That question sounds quite rhetorical.

Reply:
Maybe we should catch this where it happens, since it it sort of an invariant that we probably rely on elsewhere.

Reply:
I am all for that if it makes more sense. It seems similar to https://trac.cppcheck.net/ticket/12494 which I did not try to fix.

Reply:
If this needs a different approach but the others are fine I will drop this from the PR and a file a ticket about it instead. That way this could be merged.

Reply:
It seems we started to hit this in daca:

```
Packages: [trilinos](http://cppcheck1.osuosl.org:8000/trilinos)
0x00007ffff7d4eba4 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::compare(char const*) const () from /lib/x86_64-linux-gnu/libstdc++.so.6

#0 std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::compare(char const*) const (...) from /lib/x86_64-linux-gnu/libstdc++.so.6
#1 std::operator==<char, std::char_traits<char>, std::allocator<char> >(...) at /usr/include/c++/11/bits/basic_string.h:6248
#2  Library::isentrypoint (...) at lib/library.h:427
#3  isArrayArg (...) at build/checkautovariables.cpp:432
#4  CheckAutoVariables::autoVariables (...) at build/checkautovariables.cpp:664
#5 CheckAutoVariables::runChecks(...) at lib/checkautovariables.h:62
#6 CppCheck::checkNormalTokens(...) at build/cppcheck.cpp:1133
#7 CppCheck::checkFile(...) at build/cppcheck.cpp:966
#8 CppCheck::check(...) at build/cppcheck.cpp:567
#9 SingleExecutor::check(...) at cli/singleexecutor.cpp:53
#10 CppCheckExecutor::check_internal(...) at cli/cppcheckexecutor.cpp:278
#11 CppCheckExecutor::check_wrapper(...) at cli/cppcheckexecutor.cpp:217
#12 CppCheckExecutor::check (...) at cli/cppcheckexecutor.cpp:203
#13 main(...) at cli/main.cpp:91
```

Reply:
I think https://trac.cppcheck.net/ticket/12501 has a different root cause though.
Can we add a nullptr check + debug message?

Reply:
Will look into it.

Are the remaining fixes fine? I would pull it out into a different PR then and merge the rest.

Reply:
Sounds good to me.

Reply:
Done: #6116.